chainntnfs/neutrinonotify/neutrino.go

```diff
@@ -307,31 +307,45 @@ func (n *NeutrinoNotifier) notificationDispatcher() {
 				currentHeight := n.bestHeight
 				n.heightMtx.RUnlock()
 
-				// Lookup whether the transaction is already included in the
-				// active chain.
-				txConf, err := n.historicalConfDetails(msg.TxID, currentHeight,
-					msg.heightHint)
-				if err != nil {
-					chainntnfs.Log.Error(err)
-				}
+				// Look up whether the transaction is already
+				// included in the active chain. We'll do this
+				// in a goroutine to prevent blocking
+				// potentially long rescans.
+				go func() {
+					confDetails, err := n.historicalConfDetails(
+						msg.TxID, currentHeight,
+						msg.heightHint,
+					)
+					if err != nil {
+						chainntnfs.Log.Error(err)
+					}
+
+					if confDetails != nil {
+						err := n.txConfNotifier.UpdateConfDetails(
+							*msg.TxID, msg.ID,
```

Comment:
Not sure if we actually need the new `ID` if we can just pass the `ConfNtfn` in here instead.

Reply:
Same comment as above w.r.t. ensuring the notification has been registered first.